Classes/Event/EnrichScriptTagEvent.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Event;
+
+use Brotkrueml\MatomoIntegration\Exceptions\InvalidDataAttributeName;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * This event allows the developer to enrich the build script tag with additional attributes.
+ */
+final class EnrichScriptTagEvent
+{
+    /**
+     * The server request in case the developer requires additional information.
+     */
+    private ServerRequestInterface $serverRequest;
+
+    /**
+     * The id of the element
+     */
+    private string $id = '';
+
+    /**
+     * The type that should be set for the script tag. I.e. text/plain
+     */
+    private string $type = '';
+
+    /**
+     * Additional data attributes.
+     * @var array <string, string|null>
+     */
+    private array $dataAttributes = [];
+
+    public function __construct(ServerRequestInterface $serverRequest)
+    {
+        $this->serverRequest = $serverRequest;
+    }
+
+    /**
+     * The server request for more context
+     */
+    public function getServerRequest(): ServerRequestInterface
+    {
+        return $this->serverRequest;
+    }
+
+    /**
+     * Set the id of script tag.
+     */
+    public function setId(string $id): void
+    {
+        $this->id = trim($id);
```

Comment:
I would not silently trim it and leave instead the value untouched (only later htmlspecialchar it). It's up to the user to give valid ids. Otherwise: We can throw an exception if in invalid/unexpected character is available in the value.

Reply:
I just removed the trim function.